### PR TITLE
feat: create_index take torch.device object

### PR DESCRIPTION
**Pull request by eddyxu**
Allow `Dataset::create_index(..., accelerator=torch.device)` so that it can takes in `lance.torch.preferred_device()`

### CODE REVIEW

File: python/python/lance/vector.py [hunk 1]
```diff
@@ -134,7 +134,9 @@ def train_ivf_centroids(
     sample_rate: int = 256,
 ) -> np.ndarray:
     """Use accelerator (GPU or MPS) to train kmeans."""
```
Comment (chebbyChefNEQ) [inline python/python/lance/vector.py:136]: nit: maybe we should rename `train_ivf_centroids_accelerator`

Comment (chebbyChefNEQ) [reply python/python/lance/vector.py:136]: also update the type signature to allow `torch.Device`

**Review by chebbyChefNEQ — APPROVED** [review]

File: python/python/lance/vector.py [hunk 2]
```diff
@@ -134,7 +134,9 @@ def train_ivf_centroids(
...
-    if not (CUDA_REGEX.match(accelerator) or accelerator == "mps"):
+    if isinstance(accelerator, str) and (
```
Comment (chebbyChefNEQ) [inline python/python/lance/vector.py:137]: I guess another nit is. I think
```python
torch.device(device)
```
should work for both string and `torch.Device` inputs.

File: python/python/lance/vector.py (hunk 1 above)
Comment (eddyxu) [reply python/python/lance/vector.py:136]: done

**Pull request merged**